src/Bicep.Cli/Helpers/ServiceCollectionExtensions.cs

```diff
@@ -40,23 +40,17 @@ public static class ServiceCollectionExtensions
     ///
     /// See https://endjin.com/blog/2020/09/simple-pattern-for-using-system-commandline-with-dependency-injection for reference.
     /// </remarks>
-    public static IServiceCollection AddCommands(this IServiceCollection services)
-    {
-        Type grabCommandType = typeof(BuildCommand);
-        Type commandType = typeof(ICommand);
-
-        IEnumerable<Type> commands = grabCommandType
-            .Assembly
-            .GetExportedTypes()
-            .Where(x => x.Namespace == grabCommandType.Namespace && x.GetInterfaces().Contains(commandType));
-
-        foreach (Type command in commands)
-        {
-            services.AddSingleton(command);
-        }
-
-        return services;
-    }
+    public static IServiceCollection AddCommands(this IServiceCollection services) =>
+        // this is harcoded to make the code trim-safe
+        // TODO: Can replace with a code generator
```

Comment:
IMO what you have here is perfect. I don't think it's too onerous to have to update this list when adding a new command! I'd argue against adding a code generator here - adding complexity for little benefit.

Reply:
That's fair.